runtime/src/main/java/org/corfudb/runtime/collections/CorfuQueue.java

```diff
@@ -36,8 +39,10 @@
      * The main CorfuTable which contains the primary key-value mappings.
      */
     private final CorfuTable<Long, E> corfuTable;
+    private final CorfuTable<Long, Long> orderTable;
```

Comment:
Can you please put comments explaining what `orderTable` and why it is needed.

Reply:
Sure. Just to let you know, we still need to confirm if this fix is necessary since the primary issue has a workaround they are ok with.